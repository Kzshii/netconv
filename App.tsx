import 'react-native-gesture-handler';
import React from 'react';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { initializeStore } from './src/store';
import { CartIcon, LogoIcon } from './src/components';
import { ProductDescriptionScreen, HomeScreen, CartScreen } from './src/pages';

const Stack = createStackNavigator();

export default function App() {
  return (
    <Provider store={initializeStore()}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen
            name="Home"
            component={HomeScreen}
            options={{
              headerStyle: { backgroundColor: '#E24E26' },
              headerTitle: () => <LogoIcon />,
              headerRight: () => <CartIcon />,
            }}
          />
          <Stack.Screen
            name="Cart"
            component={CartScreen}
            options={{
              headerStyle: { backgroundColor: '#E24E26' },
              headerTintColor: '#fff',
              headerTitle: () => <LogoIcon />,
            }}
          />
          <Stack.Screen
            name="Description"
            component={ProductDescriptionScreen}
            options={{
              headerStyle: { backgroundColor: '#E24E26' },
              headerTintColor: '#fff',
              headerTitle: () => <LogoIcon />,
              headerRight: () => <CartIcon />,
            }}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}
