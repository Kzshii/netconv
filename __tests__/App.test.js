import 'react-native';
import React from 'react';
import App from '../App.tsx';
import { shallow } from 'enzyme';

describe('<App />', () => {
  test('renders correctly App', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
