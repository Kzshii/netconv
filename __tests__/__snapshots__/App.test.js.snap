// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly App 1`] = `ShallowWrapper {}`;
